File: Backend/src/main/java/com/assignment/courseManagementSystem/services/contract/DepartmentService.java
package com.assignment.courseManagementSystem.services.contract;

import com.assignment.courseManagementSystem.dto.DepartmentDto;
import org.springframework.data.domain.Page;

public interface DepartmentService {

    DepartmentDto createDepartment(DepartmentDto dto);

    DepartmentDto getDepartmentById(Long id);

    Page<DepartmentDto> getDepartments(
            String search,
            int page,
            int size,
            String sortBy,
            String sortDir
    );

    DepartmentDto updateDepartment(Long id, DepartmentDto dto);

    void deleteDepartment(Long id);
}

File: frontend/services/departmentService.js
import api from "./api";

export const getDepartments = (params) => {
  return api.get("/api/departments", {
    params: {
      page: params.page,
      size: params.size,
      search: params.search,
      sortBy: params.sortBy,
      sortDir: params.sortDir,
    },
  });
};

export const createDepartment = (data) => {
  return api.post("/api/departments", data);
};

export const updateDepartment = (id, data) => {
  return api.put(`/api/departments/${id}`, data);
};

export const deleteDepartment = (id) => {
  return api.delete(`/api/departments/${id}`);
};
